### PR TITLE
EPAM, NodeJS practice, Homework3

### PR DESCRIPTION
**Pull request by grechintanya**

### CODE REVIEW

File: src/config/DBconnect.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import mongoose from 'mongoose';
+
+const DATABASE_URI =
```
Comment (VitalyaIlin) [inline src/config/DBconnect.ts]: it's better you use .env file for this variable

File: src/controllers/genresControllers.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  }
+};
+
+export const updateGenre = async (req: Request, res: Response) => {
+  try {
+    const id = req.params.id;
+    const updatedGenre = req.body;
+    const result = await Genre.updateOne({ _id: id }, updatedGenre);
+    if (!result.matchedCount) {
+      return res.status(404).json({ message: 'Genre not found' });
```
Comment (VitalyaIlin) [inline src/controllers/genresControllers.ts]: Handle such errors in global error handling middleware

File: src/server.ts [hunk 3]
```diff
@@ -37,6 +38,9 @@ app.use((err: Error, req: Request, res: Response, next: NextFunction) => {
   res.status(500).send('Something went wrong');
```
Comment (VitalyaIlin) [inline src/server.ts]: Handle all types of errors in this middleware, 404, 422 etc.

File: src/controllers/moviesControllers.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  if (!movies.length) {
+    return res.sendStatus(204);
+  }
+  res.status(200).json(movies);
+};
+
+export const createMovie = async (req: Request, res: Response) => {
+  try {
+    const newMovie: MovieObj = req.body;
+    const savedMovie = (await Movie.create(newMovie)).toObject();
```
Comment (VitalyaIlin) [inline src/controllers/moviesControllers.ts]: Why do you need .toObject() call here?

File: src/controllers/genresControllers.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { Request, Response } from 'express';
+
+import { Genre, GenreObj } from '../models';
+import { errorHandler } from '../middleware/errorHandlers';
+
+export const getAllGenres = async (req: Request, res: Response) => {
+  const genres: GenreObj[] = await Genre.find();
+  if (!genres.length) {
+    return res.sendStatus(204);
```
Comment (VitalyaIlin) [inline src/controllers/genresControllers.ts]: In such cases, it is more appropriate to send an empty array with a status code of 200
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

**Review by VitalyaIlin — CHANGES_REQUESTED** [review]

**Comment by grechintanya** [comment]
I've remade error handling. Please, have a look

**Review by VitalyaIlin — APPROVED** [review]

**Pull request merged**